cirq/ops/three_qubit_gates.py

```diff
@@ -145,6 +144,125 @@ def __str__(self) -> str:
         return 'CCZ**{}'.format(self._exponent)
 
 
+class ThreeQubitDiagonalGate(gate_features.ThreeQubitGate):
+    """A gate given by a diagonal 8x8 matrix."""
+
+    def __init__(self,
+                 a: float = 0.0,
+                 b: float = 0.0,
+                 c: float = 0.0,
+                 d: float = 0.0,
```

Comment:
We should include support for sympy expressions in addition to floats. This involves making `_has_unitary_` return False, `_unitary_` return NotImplemented etc when one of the angles is symbolic. Also you'd add `_resolve_parameters_` / `_is_parameterized_` methods.

Reply:
OK, so each entry may be a sympy expression or a float?

Reply:
Yes